R/list_of.R

```diff
@@ -65,6 +65,15 @@ new_list_of <- function(x = list(), ptype = logical(), ..., class = character())
   new_vctr(x, ..., ptype = ptype, class = c(class, "vctrs_list_of"))
 }
 
+#' @export
+#' @rdname list_of
+validate_list_of <- function(x) {
+  # Type checking is a side effect of the constructor
+  list_of(!!!x, .ptype = attr(x, "ptype"))
```

Comment:
Why not just inline the code and do `map(x, vec_cast, to = ptype)`?  Also the code currently doesn't check that `x` is a list